merqube_client_lib/api_client/merqube_client.py

```diff
@@ -3,15 +3,107 @@
 """
 from typing import Any
 
+import pandas as pd
 from cachetools import LRUCache, cached
 
+from merqube_client_lib.api_client.base import MerqubeApiClientBase
 from merqube_client_lib.api_client.indexapi import IndexAPIClient
+from merqube_client_lib.pydantic_types import IndexDefinitionPatchPutGet as Index
 from merqube_client_lib.secapi.client import SecAPIClient
 from merqube_client_lib.session import MerqubeAPISession
+from merqube_client_lib.types import Manifest
 
 
 class MerqubeAPIClient(IndexAPIClient, SecAPIClient):
-    """combined Merqube client"""
+    """
+    Combined API Client for indexapi + secapi, for methods that do not pertain to a particular index
+    (general queries, CRUD of indices, etc)
+    For a client pertaining to a specific existing index, use MerqubeAPIClientSingleIndex
+    """
+
+
+class MerqubeAPIClientSingleIndex(MerqubeApiClientBase):
```

Comment:
Wondering about the expected usage pattern here. Is this based on our current internal usage patterns or expected client usage patterns?

How will these clients hang together? Will `MerqubeAPIClient` have a `get_index_client` that returns a `MerqubeAPIClientSingleIndex`? Or will we need to get the name and build `MerqubeAPIClientSingleIndex` manually?

Reply:
this is a good question, not quite sure. 

my thinking was:
- some operations are index-agnostic. bulk secapi queries, and index crud (eg creating a new index) for example
- some operations are for "my [customer] index" - getting returns, other metrics, etc. you obv have to tell it what index you want, 

I was thinking the first is served by APIClient and the latter is served by SingleIndex

however, I am definitely open to a different organization.

or, maybe SingleIndex should inherit from APIClient?

WDYT?

Reply:
I think it sounds reasonable. Even if a client has multiple indices, they will modify them one at a time (unless we support bulk updates). Having SingleIndex in its own object will make it easier to understand from looking at the API what methods apply to a single index

Reply:
thanks - ill add some docs about this in the coming pr.